cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/common/StructuredRecordToCubeFact.java

```diff
@@ -126,63 +83,98 @@ public CubeFact transform(StructuredRecord record) throws Exception {
 
   private static final class CubeFactBuilder {
     private final TimestampResolver timestampResolver;
-    private final Collection<DimensionValueResolver> dimensionResolvers;
     private final Collection<MeasurementResolver> measurementResolvers;
 
-    public CubeFactBuilder(MappingConfig config) {
-      this.timestampResolver = new TimestampResolver(config.timestamp);
-      if (config.dimensions == null) {
-        throw new IllegalArgumentException("Missing required 'dimensions' configuration: " + config);
-      }
-      this.dimensionResolvers = Lists.newArrayList();
-      for (ValueMapping mapping : config.dimensions) {
-        dimensionResolvers.add(new DimensionValueResolver(mapping));
+    public CubeFactBuilder(Map<String, String> properties) {
+      Map<String, String> props = new HashMap<>(properties);
+      this.timestampResolver = new TimestampResolver(props);
+      this.measurementResolvers = Lists.newArrayList();
+      addMeasurementsFromProperty(props);
+      for (Map.Entry<String, String> property : props.entrySet()) {
+        if (property.getKey().startsWith(Properties.Cube.MEASUREMENT_PREFIX)) {
+          measurementResolvers.add(new MeasurementResolver(property.getKey(), property.getValue()));
+        }
       }
-      if (config.measurements == null) {
-        throw new IllegalArgumentException("Missing required 'measurements' configuration: " + config);
+      if (measurementResolvers.isEmpty()) {
```

Comment:
So we will throw these exceptions only after the adapter is started? Is it possible to do these validations during adapter configuration time. DBSource#configurePipeline is a good example.